examples/audio/melgan_spectrogram_inversion.py

```diff
@@ -0,0 +1,619 @@
+        )
+        self.convs.append(
+            addon_layers.WeightNormalization(
+                Conv1D(64, 41, 4, "same", groups=4), data_init=False
+            )
+        )
+        self.convs.append(
+            addon_layers.WeightNormalization(
+                Conv1D(256, 41, 4, "same", groups=16), data_init=False
+            )
+        )
+        self.convs.append(
+            addon_layers.WeightNormalization(
+                Conv1D(1024, 41, 4, "same", groups=64), data_init=False
+            )
+        )
+        self.convs.append(
+            addon_layers.WeightNormalization(
+                Conv1D(1024, 41, 4, "same", groups=256), data_init=False
+            )
+        )
+        self.convs.append(
+            addon_layers.WeightNormalization(
+                Conv1D(1024, 5, 1, "same"), data_init=False
+            )
+        )
+        self.convs.append(
+            addon_layers.WeightNormalization(Conv1D(1, 3, 1, "same"), data_init=False)
+        )
+        self.lrelu = LeakyReLU
+
+    def call(self, input):
+        out_list = []
+        for index, conv in enumerate(self.convs):
+            out = conv(input if index == 0 else out)
+            if index != len(self.convs) - 1:
+                out = self.lrelu()(out)
+            out_list.append(out)
+        return out_list
+
+
+"""
+### Create the generator
+"""
+
+
+def create_generator(input_shape):
+    inp = Input(input_shape)
+    x = MelSpec()(inp)
+    x = Conv1D(512, 7, padding="same")(x)
+    x = LeakyReLU()(x)
+    x = ConvBlock(256, 8)(x)
+    x = ConvBlock(128, 8)(x)
+    x = ConvBlock(64, 2)(x)
+    x = ConvBlock(32, 2)(x)
+    x = addon_layers.WeightNormalization(
+        Conv1D(1, 7, padding="same", activation="tanh")
+    )(x)
+    return tf.keras.models.Model(inp, x)
+
+
+# We use a dynamic input shape for the generator since the model is fully convolutional
+generator = create_generator((None, 1))
+generator.summary()
+tf.keras.utils.plot_model(generator, show_shapes=True)
+
+"""
+### Create the discriminator
+"""
+
+
+def create_discriminator(input_shape):
+    inp = Input(input_shape)
+    out_map1 = DiscriminatorBlock()(inp)
+    pool1 = AveragePooling1D()(inp)
+    out_map2 = DiscriminatorBlock()(pool1)
+    pool2 = AveragePooling1D()(pool1)
+    out_map3 = DiscriminatorBlock()(pool2)
+    return tf.keras.models.Model(inp, [out_map1, out_map2, out_map3])
+
+
+# We use a dynamic input shape for the discriminator
+# This is done because the input shape for the generator is unknown
+discriminator = create_discriminator((None, 1))
+
+discriminator.summary()
+tf.keras.utils.plot_model(discriminator, show_shapes=True)
+
+"""
+## Defining the loss functions
+
+**Generator Loss**
+
+The generator architecture uses a combination of two losses
+
+1. MeanSquaredError:
+
+This is the standard MSE generator loss calculated between ones and the outputs from the
+discriminator with _N_ layers.
+
+\begin{align}
```

Comment:
No latex notation, see guide https://github.com/keras-team/keras-io/blob/master/contributor_guide.md

Reply:
I have added images instead as advised by the guidelines. If you want me to remove the equations entirely then please let me know and I will remove them.